src/core/lib/gprpp/fork.cc

```diff
@@ -161,24 +162,13 @@ class ThreadState {
   int count_;
 };
 
-}  // namespace internal
+}  // namespace
 
 void Fork::GlobalInit() {
   if (!override_enabled_) {
     support_enabled_.store(GPR_GLOBAL_CONFIG_GET(grpc_enable_fork_support),
                            std::memory_order_relaxed);
   }
-  if (support_enabled_.load(std::memory_order_relaxed)) {
-    exec_ctx_state_ = new internal::ExecCtxState();
-    thread_state_ = new internal::ThreadState();
-  }
-}
-
-void Fork::GlobalShutdown() {
```

Comment:
I believe this deletion broadens the impact for some kinds of bugs, which is maybe fine given the ultimate goal for no-shutdown, but it should be noted. For instance, it may be the case already that gRPC sometimes shuts down with nonzero state counters. This would be a preexisting bug if it happened, but would have limited impact since those counters were reset on subsequent init/shutdown cycles. So with this change, applications _may_ see failures to fork in scenarios where they didn't before, if they were going through multiple init/shutdown cycles.